### PR TITLE
BUGFIX: SD-526 fix treemap with null data point; SD-525 fix disable saturation others series in combo

### PR DESCRIPTION
**Pull request by lanhq**
[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2320
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2077

# Related Jira tasks

- SD-525: https://jira.intgdc.com/browse/SD-525
- SD-526: https://jira.intgdc.com/browse/SD-526

### CODE REVIEW

**Comment by lanhq** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1583/ : ABORTED in 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5200/ : SUCCESS in 11s

**Comment by lanhq** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4089/ : FAILURE in 8m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5199/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2224/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1584/ : SUCCESS in 13m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5201/ : SUCCESS in 16s

**Comment by lanhq** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1046.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1587/ : SUCCESS in 14m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5207/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4092/ : FAILURE in 17m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5206/ : SUCCESS in 10s

**Comment by lanhq** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4093/ : SUCCESS in 24m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5208/ : SUCCESS in 11s

File: src/components/visualizations/chart/highcharts/comboConfiguration.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import { LINE_WIDTH } from "./lineConfiguration";
 import { IChartConfig } from "../../../../interfaces/Config";
 import { VisualizationTypes } from "../../../../constants/visualizationTypes";
 import { isLineChart } from "../../utils/common";
+import { isBucketEmpty } from "../../../../helpers/MdObjectHelper";
+import { VisualizationObject } from "@gooddata/typings";
```
Comment (nadtan) [inline src/components/visualizations/chart/highcharts/comboConfiguration.ts]: Please move this line to top

File: src/components/visualizations/chart/highcharts/comboConfiguration.ts [hunk 2]
```diff
@@ -28,13 +31,22 @@ export function getDefaultChartType(config?: IChartConfig) {
     return LINE;
 }
 
-function isLineLineCombo(config: IChartConfig): boolean {
+function isOnlyLineSeries(config: IChartConfig): boolean {
     const { primaryChartType, secondaryChartType } = getDefaultComboTypes(config);
-    return isLineChart(primaryChartType) && isLineChart(secondaryChartType);
+    const mdObj: VisualizationObject.IVisualizationObjectContent = get(config, "mdObject");
```
Comment (nadtan) [inline src/components/visualizations/chart/highcharts/comboConfiguration.ts]: Should get `buckets` here and pass it to `isBucketEmpty()`
```
const buckets: VisualizationObject.IBucket[] = get(config, "mdObject.buckets", []);
```

File: src/components/visualizations/chart/highcharts/test/comboConfiguration.spec.ts [hunk 3]
```diff
@@ -39,7 +76,7 @@ describe("Combo Configuration", () => {
         });
     });
 
-    it("should disable saturation on other line series in line_line combo chart?", () => {
+    it("should disable saturation on other series in combo which have line series on both axis", () => {
```
Comment (naibinh) [inline src/components/visualizations/chart/highcharts/test/comboConfiguration.spec.ts]: --> `should disable saturation on other series in combo which has line series on both axes`

File: src/helpers/MdObjectHelper.ts [hunk 4]
```diff
@@ -96,10 +97,18 @@ export const mdObjectToPivotBucketProps = (
     };
 };
 
+export function isBucketEmpty(mdObject: VisualizationObject.IVisualizationObjectContent, bucketName: string) {
+    const bucket = get(mdObject, "buckets", []).find(bucket => bucket.localIdentifier === bucketName);
+
+    const bucketItems = get(bucket, "items", []);
```
[the remark this answers is not in the document]
Comment (naibinh) [reply src/helpers/MdObjectHelper.ts]: please add type to variables and return type as `boolean`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4114/ : FAILURE in 1m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5234/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4115/ : FAILURE in 5m 03s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5235/ : SUCCESS in 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4125/ : SUCCESS in 7m 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5245/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4168/ : SUCCESS in 7m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5305/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2257/ : SUCCESS in 13m 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4175/ : SUCCESS in 8m 09s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1076/ : SUCCESS in 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1288/ : SUCCESS in 36m 52s

**Pull request merged**